crates/bevy_anti_alias/src/fsr3/mod.rs

```diff
@@ -0,0 +1,549 @@
+        );
+        let jitter = wgpu_ffx::get_jitter_offset(frame_count.0 as i32, phase_count);
+        temporal_jitter.offset = Vec2::from(jitter);
+
+        // Calculate mip bias
+        let scale_factor = fsr3.quality_mode.scale_factor();
+        mip_bias.0 = -(scale_factor.log2());
+
+        // Check if we need to create or update the FSR3 context
+        let needs_new_context = match fsr3_context {
+            Some(ctx) => {
+                ctx.quality_mode != fsr3.quality_mode
+                    || ctx.max_upscale_size != [upscale_resolution.x, upscale_resolution.y]
+            }
+            None => true,
+        };
+
+        if needs_new_context {
+            // Calculate maximum render size based on quality mode
+            let max_render_size = fsr3.quality_mode.render_resolution(upscale_resolution);
+
+            // Setup FSR3 context flags
+            let mut flags = FsrContextFlags::HIGH_DYNAMIC_RANGE
+                | FsrContextFlags::DEPTH_INVERTED
+                | FsrContextFlags::DEPTH_INFINITE;
+
+            // Check if using infinite depth by examining the projection matrix
+            // For infinite far plane with reversed z: clip_from_view[3][2] == near and far == infinity
+            // We check if w_axis.w == 0.0 (perspective) and assume infinite if so
+            let clip_from_view = view.clip_from_view;
+            if clip_from_view.w_axis.w == 0.0 {
+                // Perspective projection - assume infinite depth for now
+                // (Bevy typically uses infinite reversed-z for perspective)
+                flags |= FsrContextFlags::DEPTH_INFINITE;
+            }
+
+            // Create FSR3 context
+            let context_info = FsrContextInfo {
+                device: render_device.wgpu_device().clone(),
+                queue: wgpu::Queue::clone(&render_queue),
+                max_render_size: [max_render_size.x, max_render_size.y],
+                max_upscale_size: [upscale_resolution.x, upscale_resolution.y],
+                flags,
+            };
+
+            let context = FsrContext::new(context_info);
+
+            commands.entity(entity).insert((
+                Fsr3RenderContext {
+                    context: Mutex::new(context),
+                    quality_mode: fsr3.quality_mode,
+                    max_upscale_size: [upscale_resolution.x, upscale_resolution.y],
+                },
+                MainPassResolutionOverride(render_resolution),
+            ));
+        } else {
+            // Update resolution override in case it changed
+            commands
+                .entity(entity)
+                .insert(MainPassResolutionOverride(render_resolution));
+        }
+    }
+}
+
+fn prepare_fsr3_textures(
+    mut commands: Commands,
+    mut texture_cache: ResMut<TextureCache>,
+    render_device: Res<RenderDevice>,
+    views: Query<(Entity, &ExtractedCamera, &ExtractedView, &Fsr3), With<Fsr3>>,
+) {
+    for (entity, camera, view, fsr3) in &views {
```

Comment:
Please follow the newer pattern for texture-prepare systems as used in https://github.com/bevyengine/bevy/blob/main/crates/bevy_solari/src/realtime/prepare.rs (you can just store TextureView now as textures can be retrieved from views, I should update the solari code)